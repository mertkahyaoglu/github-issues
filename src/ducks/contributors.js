export const CONTRIBUTORS_REQUEST = 'contributors/REQUEST';
export const CONTRIBUTORS_SUCCESS = 'contributors/SUCCESS';
export const CONTRIBUTORS_FAILED = 'contributors/FAILED';

const initialState = {
  loading: false,
  error: null,
  contributors: [],
};

export default function reducer(state = initialState, { type, payload }) {
  switch (type) {
    case CONTRIBUTORS_REQUEST:
      return {
        ...state,
        loading: true,
      };
    case CONTRIBUTORS_FAILED:
      return {
        ...state,
        error: payload,
        loading: false,
      };
    case CONTRIBUTORS_SUCCESS:
      return {
        ...state,
        contributors: payload,
        loading: false,
      };
    default:
      return state;
  }
}

export const API_URL =
  'https://api.github.com/repos/jekyll/jekyll-admin/contributors';

export const getContributors = () => async dispatch => {
  dispatch({ type: CONTRIBUTORS_REQUEST });
  try {
    const response = await fetch(API_URL);
    const data = await response.json();
    dispatch({ type: CONTRIBUTORS_SUCCESS, payload: data });
  } catch (error) {
    console.log(error);
    dispatch({ type: CONTRIBUTORS_FAILED, payload: error });
  }
};
