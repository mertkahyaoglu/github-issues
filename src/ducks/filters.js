export const FILTER_AUTHOR = 'filters/FILTER_AUTHOR';
export const REMOVE_AUTHOR = 'filters/REMOVE_AUTHOR';
export const FILTER_LABEL = 'filters/FILTER_LABEL';
export const REMOVE_LABEL = 'filters/REMOVE_LABEL';

const initialState = {
  filteredAuthor: null,
  filteredLabel: null,
};

export default function reducer(state = initialState, { type, payload }) {
  switch (type) {
    case FILTER_AUTHOR:
      return {
        ...state,
        filteredAuthor: payload,
      };
    case FILTER_LABEL:
      return {
        ...state,
        filteredLabel: payload,
      };
    case REMOVE_AUTHOR:
      return {
        ...state,
        filteredAuthor: null,
      };
    case REMOVE_LABEL:
      return {
        ...state,
        filteredLabel: null,
      };
    default:
      return state;
  }
}

export const filterAuthor = id => ({
  type: FILTER_AUTHOR,
  payload: id,
});

export const filterLabel = id => ({
  type: FILTER_LABEL,
  payload: id,
});

export const removeAuthor = () => ({
  type: REMOVE_AUTHOR,
});

export const removeLabel = () => ({
  type: REMOVE_LABEL,
});
