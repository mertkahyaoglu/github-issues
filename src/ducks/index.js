export { default as issues } from './issues';
export { default as contributors } from './contributors';
export { default as labels } from './labels';
export { default as filters } from './filters';
