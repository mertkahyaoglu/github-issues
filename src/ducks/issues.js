export const ISSUES_REQUEST = 'issues/REQUEST';
export const ISSUES_SUCCESS = 'issues/SUCCESS';
export const ISSUES_FAILED = 'issues/FAILED';

const initialState = {
  loading: false,
  error: null,
  issues: [],
};

export default function reducer(state = initialState, { type, payload }) {
  switch (type) {
    case ISSUES_REQUEST:
      return {
        ...state,
        loading: true,
      };
    case ISSUES_FAILED:
      return {
        ...state,
        error: payload,
        loading: false,
      };
    case ISSUES_SUCCESS:
      return {
        ...state,
        issues: payload,
        loading: false,
      };
    default:
      return state;
  }
}

export const API_URL =
  'https://api.github.com/repos/jekyll/jekyll-admin/issues';

export const getIssues = () => async dispatch => {
  dispatch({ type: ISSUES_REQUEST });
  try {
    const response = await fetch(API_URL);
    const data = await response.json();
    dispatch({ type: ISSUES_SUCCESS, payload: data });
  } catch (error) {
    console.log(error);
    dispatch({ type: ISSUES_FAILED, payload: error });
  }
};

const issuesSelector = state => state.issues;

export const issuesByAuthorSelector = (state, author) => {
  const issues = issuesSelector(state);
  const selectedIssues = author
    ? issues.filter(issue => issue.user.id === author.id)
    : issues;
  return {
    ...state,
    issues: selectedIssues,
  };
};

export const issuesByLabelSelector = (state, label) => {
  const issues = issuesSelector(state);
  const selectedIssues = label
    ? issues.filter(issue => issue.labels.find(l => l.id === label.id))
    : issues;
  return {
    ...state,
    issues: selectedIssues,
  };
};
