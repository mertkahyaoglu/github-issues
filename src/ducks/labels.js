export const LABELS_REQUEST = 'labels/REQUEST';
export const LABELS_SUCCESS = 'labels/SUCCESS';
export const LABELS_FAILED = 'labels/FAILED';

const initialState = {
  loading: false,
  error: null,
  labels: [],
};

export default function reducer(state = initialState, { type, payload }) {
  switch (type) {
    case LABELS_REQUEST:
      return {
        ...state,
        loading: true,
      };
    case LABELS_FAILED:
      return {
        ...state,
        error: payload,
        loading: false,
      };
    case LABELS_SUCCESS:
      return {
        ...state,
        labels: payload,
        loading: false,
      };
    default:
      return state;
  }
}

export const API_URL =
  'https://api.github.com/repos/jekyll/jekyll-admin/labels';

export const getLabels = () => async dispatch => {
  dispatch({ type: LABELS_REQUEST });
  try {
    const response = await fetch(API_URL);
    const data = await response.json();
    dispatch({ type: LABELS_SUCCESS, payload: data });
  } catch (error) {
    console.log(error);
    dispatch({ type: LABELS_FAILED, payload: error });
  }
};
