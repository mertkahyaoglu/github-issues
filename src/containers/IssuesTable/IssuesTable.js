import React, { Component } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import Check from 'react-icons/lib/fa/check';
import ExclamationIcon from 'react-icons/lib/fa/exclamation-circle';
import ArrowDown from 'react-icons/lib/fa/caret-down';
import Status from 'components/Status';
import IssueRow from 'components/IssueRow';
import Dropdown from 'components/Dropdown';
import ClickOutside from 'react-click-outside';
import { Avatar } from 'styles';
import {
  getIssues,
  issuesByAuthorSelector,
  issuesByLabelSelector,
} from 'ducks/issues';
import { getContributors } from 'ducks/contributors';
import { getLabels } from 'ducks/labels';
import { filterAuthor, filterLabel } from 'ducks/filters';

export class IssuesTable extends Component {
  state = {
    authorDropdownOpen: false,
    labelDropdownOpen: false,
  };

  static propTypes = {
    data: PropTypes.shape({
      issues: PropTypes.array,
      loading: PropTypes.bool,
      error: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
    }).isRequired,
  };

  componentDidMount() {
    const { getIssues } = this.props;
    getIssues();
  }

  countIssues = issues => {
    const open = issues.filter(issue => issue.state === 'open').length;
    const closed = issues.length - open;
    return {
      open,
      closed,
    };
  };

  openAuthorsDropdown = () => {
    const { authorDropdownOpen } = this.state;
    const { getContributors, contributors: { contributors } } = this.props;
    this.setState({ authorDropdownOpen: !authorDropdownOpen });
    if (contributors.length === 0) {
      getContributors();
    }
  };

  closeAuthorsDropdown = () => {
    this.setState({ authorDropdownOpen: false });
  };

  openLabelsDropdown = () => {
    const { labelDropdownOpen } = this.state;
    const { getLabels, labels: { labels } } = this.props;
    this.setState({ labelDropdownOpen: !labelDropdownOpen });
    if (labels.length === 0) {
      getLabels();
    }
  };

  closeLabelsDropdown = () => {
    this.setState({ labelDropdownOpen: false });
  };

  renderIssueRows = issues =>
    issues.map(issue => <IssueRow key={issue.id} data={issue} />);

  filterByAuthor = author => {
    this.closeAuthorsDropdown();
    this.props.filterAuthor(author);
  };

  filterByLabel = label => {
    this.closeLabelsDropdown();
    this.props.filterLabel(label);
  };

  render() {
    const { authorDropdownOpen, labelDropdownOpen } = this.state;
    const {
      data,
      contributors: { contributors },
      labels: { labels },
      loading,
      error,
    } = this.props;

    if (loading) {
      return 'Loading';
    }

    if (error) {
      return 'Error';
    }

    const counts = this.countIssues(data.issues);
    return (
      <div>
        <Header>
          <IssuesCount>
            <StyledStatus icon={ExclamationIcon}>
              {counts.open} Open
            </StyledStatus>
            <StyledStatus icon={Check}>{counts.closed} Closed</StyledStatus>
          </IssuesCount>
          <ButtonsWrapper>
            <Button onClick={this.openAuthorsDropdown}>
              Author <ArrowDown />
              {contributors.length > 0 &&
                authorDropdownOpen && (
                  <ClickOutside onClickOutside={this.closeAuthorsDropdown}>
                    <Dropdown
                      isOpen={authorDropdownOpen}
                      data={contributors}
                      title="Sort by author"
                      onItemClick={item => this.filterByAuthor(item)}
                      renderListItem={item => {
                        return (
                          <ListItem>
                            <Avatar src={item.avatar_url} />
                            <LoginName>{item.login}</LoginName>
                          </ListItem>
                        );
                      }}
                    />
                  </ClickOutside>
                )}
            </Button>
            <Button onClick={this.openLabelsDropdown}>
              Label <ArrowDown />
              {labels.length > 0 &&
                labelDropdownOpen && (
                  <ClickOutside onClickOutside={this.closeLabelsDropdown}>
                    <Dropdown
                      isOpen={labelDropdownOpen}
                      data={labels}
                      title="Sort by label"
                      onItemClick={item => this.filterByLabel(item)}
                      renderListItem={item => {
                        return (
                          <ListItem>
                            <Color color={item.color} />
                            <span>{item.name}</span>
                          </ListItem>
                        );
                      }}
                    />
                  </ClickOutside>
                )}
            </Button>
          </ButtonsWrapper>
        </Header>
        <RowsWrapper>{this.renderIssueRows(data.issues)}</RowsWrapper>
      </div>
    );
  }
}

const mapStateToProps = state => {
  const {
    filters: { filteredAuthor, filteredLabel },
    issues,
    contributors,
    labels,
  } = state;
  const issuesFilteredByAuthor = issuesByAuthorSelector(issues, filteredAuthor);
  const issuesFilteredByLabel = issuesByLabelSelector(
    issuesFilteredByAuthor,
    filteredLabel
  );
  return {
    data: issuesFilteredByLabel,
    contributors,
    labels,
  };
};

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      getIssues,
      getContributors,
      getLabels,
      filterAuthor,
      filterLabel,
    },
    dispatch
  );

export default connect(mapStateToProps, mapDispatchToProps)(IssuesTable);

const Header = styled.div`
  position: relative;
  background-color: #f6f8fa;
  border: 1px solid #e1e4e8;
  border-radius: 3px 3px 0 0;
  padding: 13px;
`;

const RowsWrapper = styled.div`
  border-left: 1px #e1e4e8 solid;
  border-right: 1px #e1e4e8 solid;
`;

const IssuesCount = styled.div`
  display: inline-flex;
`;

const ButtonsWrapper = styled.div`
  float: right;
`;

const StyledStatus = styled(Status)`
  margin-right: 10px;
`;

const Button = styled.button`
  background: none;
  border: none;
  white-space: nowrap;
  user-select: none;
  cursor: pointer;
  position: relative;
  outline: none;
`;

const ListItem = styled.div`
  display: flex;
  align-items: center;
`;

const LoginName = styled.span`
  margin-left: 4px;
  color: #000;
`;

const Color = styled.div`
  margin-right: 4px;
  margin-top: -1px;
  background-color: #${props => props.color};
  display: inline-block;
  width: 14px;
  height: 14px;
  vertical-align: middle;
  border-radius: 3px;
`;
