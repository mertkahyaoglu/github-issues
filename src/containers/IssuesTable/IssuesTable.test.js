import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
import { IssuesTable } from './IssuesTable';
import { FILTER_AUTHOR, filterAuthor, removeAuthor } from 'ducks/filters';

Enzyme.configure({ adapter: new Adapter() });

describe('IssuesTable', () => {
  let wrapper, store, props;

  beforeEach(() => {
    props = {
      getIssues: jest.fn(),
      getContributors: jest.fn(),
      getLabels: jest.fn(),
      filterAuthor: jest.fn(),
      filterLabel: jest.fn(),
      data: {
        issues: [mockIssue],
        loading: false,
        error: null,
      },
      contributors: {
        contributors: [mockContributor],
        loading: false,
        error: null,
      },
      labels: {
        labels: [mockLabel],
        loading: false,
        error: null,
      },
    };

    wrapper = shallow(<IssuesTable {...props} />);
  });

  it('capturing snapshot of IssuesTable', () => {
    const tree = renderer.create(<IssuesTable {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('check actions on dispatching when mounted', () => {
    expect(props.getIssues.mock.calls.length).toBe(1);
  });
});

const mockIssue = {
  url: 'https://api.github.com/repos/jekyll/jekyll-admin/issues/456',
  repository_url: 'https://api.github.com/repos/jekyll/jekyll-admin',
  labels_url:
    'https://api.github.com/repos/jekyll/jekyll-admin/issues/456/labels{/name}',
  comments_url:
    'https://api.github.com/repos/jekyll/jekyll-admin/issues/456/comments',
  events_url:
    'https://api.github.com/repos/jekyll/jekyll-admin/issues/456/events',
  html_url: 'https://github.com/jekyll/jekyll-admin/issues/456',
  id: 307287639,
  number: 456,
  title: 'Security issues',
  user: {
    login: 'theunluckyone1',
    id: 37171438,
    avatar_url: 'https://avatars2.githubusercontent.com/u/37171438?v=4',
    gravatar_id: '',
    url: 'https://api.github.com/users/theunluckyone1',
    html_url: 'https://github.com/theunluckyone1',
    followers_url: 'https://api.github.com/users/theunluckyone1/followers',
    following_url:
      'https://api.github.com/users/theunluckyone1/following{/other_user}',
    gists_url: 'https://api.github.com/users/theunluckyone1/gists{/gist_id}',
    starred_url:
      'https://api.github.com/users/theunluckyone1/starred{/owner}{/repo}',
    subscriptions_url:
      'https://api.github.com/users/theunluckyone1/subscriptions',
    organizations_url: 'https://api.github.com/users/theunluckyone1/orgs',
    repos_url: 'https://api.github.com/users/theunluckyone1/repos',
    events_url: 'https://api.github.com/users/theunluckyone1/events{/privacy}',
    received_events_url:
      'https://api.github.com/users/theunluckyone1/received_events',
    type: 'User',
    site_admin: false,
  },
  labels: [],
  state: 'open',
  locked: false,
  assignee: null,
  assignees: [],
  milestone: null,
  comments: 0,
  created_at: '2018-03-21T15:13:00Z',
  updated_at: '2018-03-21T15:13:00Z',
  closed_at: null,
  author_association: 'NONE',
};

const mockContributor = {
  login: 'mertkahyaoglu',
  id: 7414026,
  avatar_url: 'https://avatars0.githubusercontent.com/u/7414026?v=4',
  gravatar_id: '',
  url: 'https://api.github.com/users/mertkahyaoglu',
  html_url: 'https://github.com/mertkahyaoglu',
  followers_url: 'https://api.github.com/users/mertkahyaoglu/followers',
  following_url:
    'https://api.github.com/users/mertkahyaoglu/following{/other_user}',
  gists_url: 'https://api.github.com/users/mertkahyaoglu/gists{/gist_id}',
  starred_url:
    'https://api.github.com/users/mertkahyaoglu/starred{/owner}{/repo}',
  subscriptions_url: 'https://api.github.com/users/mertkahyaoglu/subscriptions',
  organizations_url: 'https://api.github.com/users/mertkahyaoglu/orgs',
  repos_url: 'https://api.github.com/users/mertkahyaoglu/repos',
  events_url: 'https://api.github.com/users/mertkahyaoglu/events{/privacy}',
  received_events_url:
    'https://api.github.com/users/mertkahyaoglu/received_events',
  type: 'User',
  site_admin: false,
  contributions: 810,
};

const mockLabel = {
  id: 549257748,
  url: 'https://api.github.com/repos/jekyll/jekyll-admin/labels/accepted',
  name: 'accepted',
  color: '4bc865',
  default: false,
};
