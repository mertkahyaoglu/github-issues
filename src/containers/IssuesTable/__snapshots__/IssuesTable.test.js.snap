// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssuesTable capturing snapshot of IssuesTable 1`] = `
<div>
  <div
    className="sc-chPdSV eGHSRH"
  >
    <div
      className="sc-kGXeez dafFID"
    >
      <div
        className="sc-dxgOiQ OhWjr sc-bdVaJa xDNXI"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "marginRight": "0.5em",
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m20.1 2.9q4.7 0 8.6 2.3t6.3 6.2 2.3 8.6-2.3 8.6-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3z m2.9 27.8v-4.2q0-0.4-0.2-0.6t-0.5-0.2h-4.3q-0.3 0-0.5 0.2t-0.2 0.5v4.3q0 0.3 0.2 0.5t0.5 0.2h4.3q0.3 0 0.5-0.2t0.2-0.5z m0-7.7l0.4-13.8q0-0.3-0.3-0.5-0.2-0.1-0.5-0.1h-4.9q-0.3 0-0.5 0.1-0.3 0.2-0.3 0.4l0.4 13.9q0 0.2 0.2 0.4t0.6 0.2h4.1q0.3 0 0.5-0.2t0.3-0.4z"
            />
          </g>
        </svg>
        1
         Open
      </div>
      <div
        className="sc-dxgOiQ OhWjr sc-bdVaJa xDNXI"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "marginRight": "0.5em",
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m37.3 12.6q0 0.9-0.6 1.6l-19.2 19.1q-0.6 0.7-1.5 0.7t-1.6-0.7l-11.1-11.1q-0.6-0.6-0.6-1.5t0.6-1.5l3.1-3q0.6-0.7 1.5-0.7t1.5 0.7l6.6 6.5 14.6-14.6q0.6-0.6 1.5-0.6t1.5 0.6l3.1 3q0.6 0.6 0.6 1.5z"
            />
          </g>
        </svg>
        0
         Closed
      </div>
    </div>
    <div
      className="sc-kpOJdX jaVOMK"
    >
      <button
        className="sc-ckVGcZ fjZFkl"
        onClick={[Function]}
      >
        Author 
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m31.4 15.7q0 0.6-0.5 1l-10 10q-0.4 0.4-1 0.4t-1-0.4l-10-10q-0.4-0.4-0.4-1t0.4-1 1-0.4h20q0.6 0 1 0.4t0.5 1z"
            />
          </g>
        </svg>
      </button>
      <button
        className="sc-ckVGcZ fjZFkl"
        onClick={[Function]}
      >
        Label 
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m31.4 15.7q0 0.6-0.5 1l-10 10q-0.4 0.4-1 0.4t-1-0.4l-10-10q-0.4-0.4-0.4-1t0.4-1 1-0.4h20q0.6 0 1 0.4t0.5 1z"
            />
          </g>
        </svg>
      </button>
    </div>
  </div>
  <div
    className="sc-kgoBCf hmeoKY"
  >
    <div
      className="sc-ifAKCX lHalS"
    >
      <div
        className="sc-bdVaJa xDNXI"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "marginRight": "0.5em",
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m20.1 2.9q4.7 0 8.6 2.3t6.3 6.2 2.3 8.6-2.3 8.6-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3z m2.9 27.8v-4.2q0-0.4-0.2-0.6t-0.5-0.2h-4.3q-0.3 0-0.5 0.2t-0.2 0.5v4.3q0 0.3 0.2 0.5t0.5 0.2h4.3q0.3 0 0.5-0.2t0.2-0.5z m0-7.7l0.4-13.8q0-0.3-0.3-0.5-0.2-0.1-0.5-0.1h-4.9q-0.3 0-0.5 0.1-0.3 0.2-0.3 0.4l0.4 13.9q0 0.2 0.2 0.4t0.6 0.2h4.1q0.3 0 0.5-0.2t0.3-0.4z"
            />
          </g>
        </svg>
      </div>
      <div
        className="sc-EHOje YINzU"
      >
        <div
          className="sc-bZQynM jdkaLI"
        >
          <a
            className="sc-gzVnrw dUTaIu"
            href="https://github.com/jekyll/jekyll-admin/issues/456"
            target="_blank"
          >
            Security issues
          </a>
          <div
            className="sc-iwsKbI bshbDz"
          />
          <div
            className="sc-gZMcBi hWXGrZ"
          />
        </div>
        <div
          className="sc-gqjmRU kNPpqd"
        >
          #456 opend on 2018-03-21T15:13:00Z by
           
          <a
            className="sc-htoDjs gjQERW"
            href="https://github.com/theunluckyone1"
            target="_blank"
          >
            theunluckyone1
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
